Fix tags in ec2_instance_facts

f3lang opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
The method boto3_tag_list_to_ansible_dict in module_utils/ec2.py changed
and does no longer check whether the returned result of boto3 uses
"key" or "Key" as the tag key identifier.
This fixes ec2_instance_facts to make this check in its own, since boto3
may return "key" instead of "Key"
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
ec2_instance_facts
##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
2.4
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

f3lang committed:
Fix tags in ec2_instance_facts
The method boto3_tag_list_to_ansible_dict in module_utils/ec2.py changed
and does no longer check whether the returned result of boto3 uses
"key" or "Key" as the tag key identifier.
This fixes ec2_instance_facts to make this check in its own, since boto3
may return "key" instead of "Key"

willthames commented on lib/ansible/modules/cloud/amazon/ec2_instance_facts.py:
```diff
@@ -500,8 +500,11 @@ def list_ec2_instances(connection, module):
     for instance in snaked_instances:
-        if 'tags' in instance:
-            instance['tags'] = boto3_tag_list_to_ansible_dict(instance['tags'])
+        if 'tags' in instance and len(instance['tags']) > 0:
```
Seems simpler to just do:

```
instance['tags'] = boto3_tag_list_to_ansible_dict(instance.get('tags', []), 'key', 'value')
```

This way, we always return `tags`. There doesn't seem to be a need to check whether `key` is in `instance['tags'][0]` as we've just snaked it (that is, even if boto3 returned `Key`, we've converted it by this point already)

f3lang replied:
Thank you for the feedback. 
You are right. Forgot about the snaking.. Made the change and pushed

f3lang committed:
Since the indices for the tags are already formatted to lowercase
by the snaking, we can assume, that the index for the tags are already
formatted

willthames commented:
Thanks very much for this!

cntrlne commented:
Any idea when this will make it into master?

willthames commented:
@jc-io there is no 'master'. 

It's in the devel branch, which is what 2.5 will be based on, it's been backported to stable-2.4, and was thus released with 2.4.1 

I'm not really sure what question you're actually asking

cntrlne commented:
Ah, me running 2.4.0 then explains it. Thx
